src/robot/libraries/String.py

```diff
@@ -125,6 +126,48 @@ def decode_bytes_to_string(self, bytes, encoding, errors='strict'):
             raise TypeError('Can not decode strings on Python 3.')
         return bytes.decode(encoding, errors)
 
+    def format_string(self, template_or_str, *posicional_search_replace,
```

Comment:
can we add unit tests?

Reply:
Yes we can, but I don't know how to do that.

Reply:
Better actually to add acceptance tests. See `atest/README.rst` for details.